cmd/handler-api.go

```diff
@@ -136,20 +136,25 @@ func maxClients(f http.HandlerFunc) http.HandlerFunc {
 			return
 		}
 
+		globalHTTPStats.addRequestsInQueue(1)
+
 		deadlineTimer := time.NewTimer(deadline)
 		defer deadlineTimer.Stop()
 
 		select {
 		case pool <- struct{}{}:
 			defer func() { <-pool }()
+			globalHTTPStats.addRequestsInQueue(-1)
```

Comment:
Defer these when the queue count is incremented.

That way even a panic will get the count properly decremented.

Reply:
> Defer these when the queue count is incremented.

It will be wrong, we need to decrement when a request is out from the pool, otherwise we won't have the exact numbers of clients in the pool.

> That way even a panic will get the count properly decremented.

panic will happen in the next line, ServerHTTP() and we already decremented at this stage.

Reply:
Make sense. Let's keep it!